client/blocks/importer/wordpress/upgrade-plan/upgrade-plan-hosting-details.tsx

```diff
@@ -2,72 +2,92 @@ import { useIsEnglishLocale } from '@automattic/i18n-utils';
 import { useState } from '@wordpress/element';
 import { hasTranslation } from '@wordpress/i18n';
 import { Icon } from '@wordpress/icons';
-import { useI18n } from '@wordpress/react-i18n';
+import { getQueryArg } from '@wordpress/url';
+import { useTranslate } from 'i18n-calypso';
+import { useHostingProviderQuery } from 'calypso/data/site-profiler/use-hosting-provider-query';
 import { UpgradePlanHostingDetailsList, UpgradePlanHostingTestimonials } from './constants';
 import { UpgradePlanHostingDetailsTooltip } from './upgrade-plan-hosting-details-tooltip';
 
 export const UpgradePlanHostingDetails = () => {
-	const { __ } = useI18n();
+	const translate = useTranslate();
 	const isEnglishLocale = useIsEnglishLocale();
 	const [ activeTooltipId, setActiveTooltipId ] = useState( '' );
+	const importSiteQueryParam = getQueryArg( window.location.href, 'from' )?.toString() || '';
+	let importSiteHostName = '';
+
+	try {
+		importSiteHostName = new URL( importSiteQueryParam )?.hostname;
+	} catch ( e ) {}
 
 	const headerMainText =
 		hasTranslation( 'Why should you host with us?' ) || isEnglishLocale
-			? __( 'Why should you host with us?' )
-			: __( 'Why you should host with us?' );
+			? translate( 'Why should you host with us?' )
+			: translate( 'Why you should host with us?' );
+
+	const { data: hostingProviderData } = useHostingProviderQuery( importSiteHostName, true );
+	const hostingProviderName = hostingProviderData?.hosting_provider?.name;
```

Comment:
I see that we have a `useHostingProviderName` ([link](https://github.com/Automattic/wp-calypso/blob/trunk/client/site-profiler/hooks/use-hosting-provider-name.ts)), can we use that?

Reply:
I added the `useHostingProviderName` to keep consistent with the logic in the site profiler.